typing-tutor/main.js

```diff
@@ -31,4 +42,16 @@ function renderAll(chars) {
   return $all
 }
 
+window.addEventListener('keydown', function () {
+  pressedKey = event.key
+  if (pressedKey !== appState.characters[appState.currentCharacter].char) {
+    characters[appState.currentCharacter].failures = 1
```

Comment:
We want to record multiple failures per character, so you'll want to increment this.